tiledb/sm/tile/generic_tile_io.cc

```diff
@@ -81,7 +82,7 @@ Status GenericTileIO::read_generic(
       GenericTileHeader::BASE_SIZE + header.filter_pipeline_size;
 
   assert(tile);
-  *tile = new Tile();
+  *tile = tdb_new(Tile);
```

Comment:
@d-hoke pointed out that this has an unmatched `tdb_delete`. My regex that I was using to match variable names was not accounting for `*` operators. I'll modify that and make another pass.

Reply:
Fixed